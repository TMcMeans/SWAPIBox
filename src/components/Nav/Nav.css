@font-face {
  font-family: 'mono';
  font-style: normal;
  font-weight: normal;
  src: url(../../anakin_mono/anakinmono.ttf);
}

/* blue color */
/* color: #228DFF; */

.nav {
  height: 40px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-content: center;
  align-items: center;
  width: 80%;
  margin: 0 auto;
  position: relative;
  z-index: 2;
  background-color: #feda4a;
  border-radius: 20px;
  top: -145px;
}

button {
  font-family: 'mono';
  /* font-family: 'Pathway Gothic One', sans-serif; */
  padding-top: 7px;
  font-size: 20px;
  width: 150px;
  height: 40px;
  color:#eabc01;
  font-weight: 600;
  background-color: #feda4a;
  border-left: 2px solid #eabc01;
  border-right: 2px solid #eabc01;
  border-top: none;
  border-bottom: none;
  -webkit-animation: neon2 1.5s ease-in-out infinite alternate;
  -moz-animation: neon2 1.5s ease-in-out infinite alternate;
  animation: neon2 1.5s ease-in-out infinite alternate;
}

button:hover {
  color: #feda4a;
  cursor: pointer;
}

@-webkit-keyframes neon2 {
  from {
    text-shadow: 0 0 10px #fff, 0 0 20px #fff, 0 0 30px #fff, 0 0 40px #feda4a, 0 0 70px #feda4a, 0 0 80px #feda4a, 0 0 100px #feda4a, 0 0 150px #feda4a;
  }
  to {
    text-shadow: 0 0 5px #fff, 0 0 10px #fff, 0 0 15px #fff, 0 0 20px #feda4a, 0 0 35px #feda4a, 0 0 40px #feda4a, 0 0 50px #feda4a, 0 0 75px #feda4a;
  }
}
