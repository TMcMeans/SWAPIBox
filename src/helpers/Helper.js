export const getRandomFilm = filmArray => {
  const randomNumber = Math.floor(Math.random() * 6);
  return filmArray[randomNumber];
};

export const cleanPeopleData = async peopleArray => {
  const people = peopleArray.map(async person => {
    const { homeworld, species } = person;

    const fetches = [fetch(homeworld), fetch(species)];
    const response = await Promise.all(fetches);

    const parsedResponses = [response[0].json(), response[1].json()];

    const parsedData = await Promise.all(parsedResponses);

    const homeworldObj = parsedData[0];
    const speciesObj = parsedData[1];

    return {
      name: person.name,
      homeworld: homeworldObj.name,
      homePop: homeworldObj.population,
      species: speciesObj.name,
      isFavorite: false
    };
  });
  return Promise.all(people);
};

export const cleanVehicleData = async vehiclesArray => {
  const vehicles = await vehiclesArray.map(async vehicle => {
    return {
      name: vehicle.name,
      model: vehicle.model,
      class: vehicle.vehicle_class,
      passengers: vehicle.passengers,
      isFavorite: false
    };
  });
  return Promise.all(vehicles);
};

export const cleanPlanetsData = async planetsArray => {
  const planets = await planetsArray.map(async planet => {
    const { residents } = planet;
    const promisedPeople = residents.map(async resident => {
      const response = await fetch(resident);
      return response.json();
    });

    const resolvedResidents = await Promise.all(promisedPeople);

    return {
      name: planet.name,
      terrain: planet.terrain,
      population: planet.population,
      climate: planet.climate,
      residents: resolvedResidents,
      isFavorite: false
    };
  });
  return Promise.all(planets);
};
