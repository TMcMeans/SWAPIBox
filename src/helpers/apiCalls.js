export const fetchFilms = async () => {
  try {
    const url = 'https://swapi.co/api/films/';
    const response = await fetch(url);
    const films = await response.json();
    return films;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPeople = async () => {
  try {
    const url = 'https://swapi.co/api/people/';
    const response = await fetch(url);
    const people = await response.json();
    return people;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchVehicles = async () => {
  try {
    const url = 'https://swapi.co/api/vehicles/';
    const response = await fetch(url);
    const vehicles = await response.json();
    return vehicles;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};

export const fetchPlanets = async () => {
  try {
    const url = 'https://swapi.co/api/planets/';
    const response = await fetch(url);
    const planets = await response.json();
    return planets;
  } catch (err) {
    console.log(`Error: ${err.message}`);
  }
};
