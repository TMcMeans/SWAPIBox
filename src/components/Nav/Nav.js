import React from 'react';
import './Nav.css';

export const Nav = () => {
  return (
    <div className="nav">
      <button>People</button>
      <button>Planets</button>
      <button>Vehicles</button>
    </div>
  );
};
