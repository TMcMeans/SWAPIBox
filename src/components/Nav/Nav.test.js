import React from 'react';
import { shallow } from 'enzyme';

import { Nav } from './Nav';
import { mockData } from '../../helpers/mockData';

describe('Nav', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const mockCardCount = 4;
    const wrapper = shallow(<Nav cardCount={mockCardCount} />);

    expect(wrapper).toMatchSnapshot();
  });
});
