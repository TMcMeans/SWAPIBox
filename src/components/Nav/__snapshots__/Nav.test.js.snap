// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    cardCount={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="nav"
        >
          <button
            className="nav-button"
          >
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#90403E",
                }
              }
              aria-current="page"
              className="link"
              name="people"
              to="/people"
            >
              PEOPLE
            </NavLink>
          </button>
          <button
            className="nav-button"
          >
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#90403E",
                }
              }
              aria-current="page"
              className="link"
              name="planets"
              to="/planets"
            >
              PLANETS
            </NavLink>
          </button>
          <button
            className="nav-button"
          >
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#90403E",
                }
              }
              aria-current="page"
              className="link"
              name="vehicles"
              to="/vehicles"
            >
              VEHICLES
            </NavLink>
          </button>
          <button
            className="nav-button"
          >
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#90403E",
                }
              }
              aria-current="page"
              className="link"
              name="favorites"
              to="/favorites"
            >
              FAVORTIES: 
              4
            </NavLink>
          </button>
        </section>,
        <section
          className="background-fade"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="people"
                to="/people"
              >
                PEOPLE
              </NavLink>
            </button>,
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="planets"
                to="/planets"
              >
                PLANETS
              </NavLink>
            </button>,
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="vehicles"
                to="/vehicles"
              >
                VEHICLES
              </NavLink>
            </button>,
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="favorites"
                to="/favorites"
              >
                FAVORTIES: 
                4
              </NavLink>
            </button>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="people"
                to="/people"
              >
                PEOPLE
              </NavLink>,
              "className": "nav-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#90403E",
                },
                "aria-current": "page",
                "children": "PEOPLE",
                "className": "link",
                "name": "people",
                "onClick": undefined,
                "to": "/people",
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="planets"
                to="/planets"
              >
                PLANETS
              </NavLink>,
              "className": "nav-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#90403E",
                },
                "aria-current": "page",
                "children": "PLANETS",
                "className": "link",
                "name": "planets",
                "onClick": undefined,
                "to": "/planets",
              },
              "ref": null,
              "rendered": "PLANETS",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="vehicles"
                to="/vehicles"
              >
                VEHICLES
              </NavLink>,
              "className": "nav-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#90403E",
                },
                "aria-current": "page",
                "children": "VEHICLES",
                "className": "link",
                "name": "vehicles",
                "onClick": undefined,
                "to": "/vehicles",
              },
              "ref": null,
              "rendered": "VEHICLES",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="favorites"
                to="/favorites"
              >
                FAVORTIES: 
                4
              </NavLink>,
              "className": "nav-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#90403E",
                },
                "aria-current": "page",
                "children": Array [
                  "FAVORTIES: ",
                  4,
                ],
                "className": "link",
                "name": "favorites",
                "onClick": undefined,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Array [
                "FAVORTIES: ",
                4,
              ],
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "background-fade",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="nav"
          >
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="people"
                to="/people"
              >
                PEOPLE
              </NavLink>
            </button>
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="planets"
                to="/planets"
              >
                PLANETS
              </NavLink>
            </button>
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="vehicles"
                to="/vehicles"
              >
                VEHICLES
              </NavLink>
            </button>
            <button
              className="nav-button"
            >
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#90403E",
                  }
                }
                aria-current="page"
                className="link"
                name="favorites"
                to="/favorites"
              >
                FAVORTIES: 
                4
              </NavLink>
            </button>
          </section>,
          <section
            className="background-fade"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="nav-button"
              >
                <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="people"
                  to="/people"
                >
                  PEOPLE
                </NavLink>
              </button>,
              <button
                className="nav-button"
              >
                <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="planets"
                  to="/planets"
                >
                  PLANETS
                </NavLink>
              </button>,
              <button
                className="nav-button"
              >
                <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="vehicles"
                  to="/vehicles"
                >
                  VEHICLES
                </NavLink>
              </button>,
              <button
                className="nav-button"
              >
                <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="favorites"
                  to="/favorites"
                >
                  FAVORTIES: 
                  4
                </NavLink>
              </button>,
            ],
            "className": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="people"
                  to="/people"
                >
                  PEOPLE
                </NavLink>,
                "className": "nav-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "activeStyle": Object {
                    "color": "#90403E",
                  },
                  "aria-current": "page",
                  "children": "PEOPLE",
                  "className": "link",
                  "name": "people",
                  "onClick": undefined,
                  "to": "/people",
                },
                "ref": null,
                "rendered": "PEOPLE",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="planets"
                  to="/planets"
                >
                  PLANETS
                </NavLink>,
                "className": "nav-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "activeStyle": Object {
                    "color": "#90403E",
                  },
                  "aria-current": "page",
                  "children": "PLANETS",
                  "className": "link",
                  "name": "planets",
                  "onClick": undefined,
                  "to": "/planets",
                },
                "ref": null,
                "rendered": "PLANETS",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="vehicles"
                  to="/vehicles"
                >
                  VEHICLES
                </NavLink>,
                "className": "nav-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "activeStyle": Object {
                    "color": "#90403E",
                  },
                  "aria-current": "page",
                  "children": "VEHICLES",
                  "className": "link",
                  "name": "vehicles",
                  "onClick": undefined,
                  "to": "/vehicles",
                },
                "ref": null,
                "rendered": "VEHICLES",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  activeStyle={
                    Object {
                      "color": "#90403E",
                    }
                  }
                  aria-current="page"
                  className="link"
                  name="favorites"
                  to="/favorites"
                >
                  FAVORTIES: 
                  4
                </NavLink>,
                "className": "nav-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "activeStyle": Object {
                    "color": "#90403E",
                  },
                  "aria-current": "page",
                  "children": Array [
                    "FAVORTIES: ",
                    4,
                  ],
                  "className": "link",
                  "name": "favorites",
                  "onClick": undefined,
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": Array [
                  "FAVORTIES: ",
                  4,
                ],
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "background-fade",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
